### ISSUE TITLE
Allow width and height change for USB input

### ISSUE THREAD

**Issue by superzanti** [hyperion-project/hyperion.ng#706]
#### What problem does this feature solve?

In the original hyperion we had the option to change the width and height of the V4L2 input. Now HyperionNG just uses the maximum resolution.

Unfortunately the maximum resolution is often very slow. I would like to divide that in half, or even quarters. Adding the sizeDecimation only slightly reduces the lag, but changing the resolution of the capture device significantly reduces the lag.

#### What does the proposed API look like?

I believe an advanced option should be added to the webserver under USB Capture for width and height. These values would then be the width and height that the VL42 USB Capture device is initialized at.

#### How should this be implemented in your opinion?

The UI elements can be added through most of the json files. Say an option called 'customWidth' and an option called 'customHeight'.

```
        // set the settings
        if (_customWidth != 0 || _customHeight != 0)
        {
                fmt.fmt.pix.width = _customWidth;
                fmt.fmt.pix.height = _customHeight;
        }
        else if (max_width != 0 || max_height != 0)
        {
                fmt.fmt.pix.width = max_width;
                fmt.fmt.pix.height = max_height;
        }
        else
        {
                fmt.fmt.pix.width = _width;
                fmt.fmt.pix.height = _height;
        }
```

Most changes would take place in hyperion/libsrc/grabber/v4l2/V4L2Grabber.cpp and associated .h files. Some changes would need to take place in JSON files to support the new UI.

#### Are you willing to work on this yourself?

Possibly, With some guidance, I already took a stab at it, and ran into an issue so I stopped. I don't think it should take a developer very long, but it would take me much longer to get familiar with the code.

**Comment by T-IOOO** [comment]
Would this solution reduce 100% CPU usage on my Rpi0?

**Comment by superzanti** [comment]
No, the divides taking place for the sizeDecimator would take up almost 0 resources.

However, cumulatively, if you reduced the resolution directly from source, it could have an impact.
Mine runs at 720x480, which is equal to 345600 pixels. Tons of operations on those pixels are then run to figure out what to put on the  LEDs. If you reduce the resolution to say 72x48, then you're decreasing all the time of those operations by a factor of 100. It just depends how prominent these functions are. But your 100% CPU usage is probably coming from elsewhere.

**Comment by T-IOOO** [comment]
I have EasyCap grabber (534d:0021) with the macrosilicon chip. My rpi0 is constantly 100% cpu. Hyperiond is peaking for sure. When I disable the grabber in the WebUI the cpu drops to 3-4%

**Comment by superzanti** [comment]
If your comfortable compiling from source just change lines 619 to 629:
```
	// set the settings
	if (max_width != 0 || max_height != 0)
	{
		fmt.fmt.pix.width = max_width;
		fmt.fmt.pix.height = max_height;
	}
	else
	{
		fmt.fmt.pix.width = _width;
		fmt.fmt.pix.height = _height;
	}
```

to the following:

```
	// set the settings
	fmt.fmt.pix.width = 160;
	fmt.fmt.pix.height = 120;
```

This quick test will show if the change works for you.

**Comment by tpmodding** [comment]
should be fixed with #766
